pkg/kube/util.go

```diff
@@ -251,7 +251,9 @@ func GetDeployMetaFromPod(pod *corev1.Pod) (metav1.ObjectMeta, metav1.TypeMeta)
 	}
 	// try to capture more useful namespace/name info for deployments, etc.
 	// TODO(dougreid): expand to enable lookup of OWNERs recursively a la kubernetesenv
-	deployMeta := pod.ObjectMeta
+
+	// Copy original metadata to avoid concurrent map writes e.g. to labels map
+	deployMeta := pod.ObjectMeta.DeepCopy()
```

Comment:
Given the expensive ManagedFields/OwnerReferences parts below, should we explicitly build a new object?

Like `deployMeta = {Name: pod.ObjectMeta.Name, Labels: pod.Objectmetata.Labels.DeepCopy()}` etc?

Reply:
Or even lazily deep-copy the labels in the replicationcontroller codepath, which must be 0.0001% of usage?

Reply:
Actually an even better idea - replace `metav1.ObjectMeta` with `types.NamespacedName`. We only ever use the name/namespace

Reply:
The `ObjectMeta` object that is returned is used as injection data in template. I think it's better not touch it. There are also several tests that check the labels. Maybe the lazy copy approach works better. It's a less intrusive change, given I'd like to backport it to previous versions..